markets/perps-market/contracts/storage/PerpsAccount.sol

```diff
@@ -249,24 +249,32 @@ library PerpsAccount {
         GlobalPerpsMarket.load().updateCollateralAmount(collateralId, amountDelta);
     }
 
-    function payDebt(Data storage self, uint256 amount) internal {
+    function payDebt(Data storage self, uint256 amount) internal returns (uint256 debtPaid) {
         if (self.debt == 0) {
             revert NonexistentDebt(self.id);
         }
 
+        /*
+            1. if the debt is less than the amount, set debt to 0 and only deposit debt amount
+            2. if the debt is more than the amount, subtract the amount from the debt
+            3. excess amount is ignored
+        */
+
         PerpsMarketFactory.Data storage perpsMarketFactory = PerpsMarketFactory.load();
+
+        if ((self.debt.toInt() - amount.toInt()) > 0) {
+            updateAccountDebt(self, -amount.toInt());
```

Comment:
np: order `debtPaid` and `updateAccountDebt` in the same order on both branches. 
In fact, if you do the assignemt before, you can do the updateAccountDebt outside of the if with just `updateAccountDebt(self, -debtPaid.toInt());`

Reply:
yep, thx.  fixed.